app/models/post.rb

```diff
@@ -257,19 +198,85 @@ def related_posts_for(user)
     end
   end
 
+  # predicates:
```

Comment:
I'm assuming these are just relocations that confused the diff (and a couple renames for top level and spam flags).  I skimmed but didn't carefully compare these to the deleted blocks.